src/include/main/database.h

```diff
@@ -131,6 +145,8 @@ class Database {
     std::shared_ptr<spdlog::logger> logger;
     std::unique_ptr<common::FileInfo> lockFile;
     std::unique_ptr<extension::ExtensionOptions> extensionOptions;
+    std::unique_ptr<DatabaseManager> databaseManager;
```

Comment:
Let's use object for DatabaseManager and StorageExtension if possible

Reply:
If we store the objet for DatabaseManager and StorageExtension we have to include the headers in the database.h.